packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts

```diff
@@ -0,0 +1,304 @@
+     * @param subMesh defines the subMesh to render
+     * @param enableAlphaMode defines if alpha mode can be changed
+     * @param effectiveMeshReplacement defines an optional mesh used to provide info for the rendering
+     * @returns the current mesh
+     */
+    public render(subMesh: SubMesh, enableAlphaMode: boolean, effectiveMeshReplacement?: AbstractMesh): Mesh {
+        const frameId = this.getScene().getFrameId();
+        if (frameId !== this._frameIdLastUpdate && this._worker && this._scene.activeCamera) {
+            this._frameIdLastUpdate = frameId;
+            this.getWorldMatrix().multiplyToRef(this._scene.activeCamera.getViewMatrix(), this._modelViewMatrix);
+            this._worker.postMessage({ view: this._modelViewMatrix.m });
+        }
+
+        return super.render(subMesh, enableAlphaMode, effectiveMeshReplacement);
+    }
+
+    /**
+     * Loads a .splat Gaussian Splatting array buffer asynchronously
+     * @param data arraybuffer containing splat file
+     * @returns a promise that resolves when the operation is complete
+     */
+
+    public loadDataAsync(data: ArrayBuffer): Promise<void> {
+        return Promise.resolve(this._loadData(data));
+    }
+
+    /**
+     * Loads a .splat Gaussian Splatting file asynchronously
+     * @param url path to the splat file to load
+     * @returns a promise that resolves when the operation is complete
+     */
+    public loadFileAsync(url: string): Promise<void> {
+        return Tools.LoadFileAsync(url, true).then((data) => {
+            this._loadData(data);
+        });
+    }
+
+    /**
+     * Releases resources associated with this mesh.
+     * @param doNotRecurse Set to true to not recurse into each children (recurse into each children by default)
+     */
+    public dispose(doNotRecurse?: boolean): void {
+        this._material.dispose(false, true);
+        this.material = null;
+
+        this._worker?.terminate();
+        this._worker = null;
+
+        super.dispose(doNotRecurse);
+    }
+
+    private static _CreateWorker = function (self: Worker) {
+        let viewProj: number[];
+        let lastProj: number[] = [];
+        let vertexCount = 0;
+        let positions: Float32Array;
+
+        self.onmessage = (e: any) => {
+            // updated on init
+            if (e.data.positions) {
+                positions = e.data.positions;
+                vertexCount = e.data.vertexCount;
+            }
+            // udpate on view changed
+            else if (e.data.view) {
+                viewProj = e.data.view;
+                const dot = lastProj[2] * viewProj[2] + lastProj[6] * viewProj[6] + lastProj[10] * viewProj[10];
+                if (!positions || Math.abs(dot - 1) < 0.01) {
+                    return;
+                }
+
+                // Sort
+                const depthMix = new BigInt64Array(vertexCount);

```

Comment:
Could we keep both (working arrays) around ?

Reply:
I mean the three of them :-) depthMix indices and floatmix ?

Reply:
I tried some things but couldn't make it work, the problem is that `depthMix` would have to be passed back and forth between the worker and the main thread if we want to share it. I think we will have to discuss about it.